### PR TITLE
Refactor ParametrizedMessage.getFormattedMessage into Strings.format

### PR DESCRIPTION
**Pull request by pgomulka**
ParametrizedMessage is part of log4j api and should not be used
in places where a simple String.format would be enough.
Many of usages like this are message formatting for exceptions being
thrown.

relates #86549

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/SingleNodeShutdownMetadata.java [hunk 1]
```diff
@@ -113,11 +114,11 @@ private SingleNodeShutdownMetadata(
         this.allocationDelay = allocationDelay;
         if (targetNodeName != null && type != Type.REPLACE) {
             throw new IllegalArgumentException(
-                new ParameterizedMessage(
-                    "target node name is only valid for REPLACE type shutdowns, " + "but was given type [{}] and target node name [{}]",
+                format(
+                    "target node name is only valid for REPLACE type shutdowns, " + "but was given type [%s] and target node name [%s]",
```
Comment (grcevski) [inline server/src/main/java/org/elasticsearch/cluster/metadata/SingleNodeShutdownMetadata.java:118]: We can get rid of the `+` here, it seems the message fits on one line.

File: x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/PutShutdownNodeAction.java [hunk 2]
```diff
@@ -156,11 +156,11 @@ public ActionRequestValidationException validate() {
 
             if (targetNodeName != null && type != SingleNodeShutdownMetadata.Type.REPLACE) {
                 arve.addValidationError(
-                    new ParameterizedMessage(
-                        "target node name is only valid for REPLACE type shutdowns, " + "but was given type [{}] and target node name [{}]",
+                    format(
+                        "target node name is only valid for REPLACE type shutdowns, " + "but was given type [%s] and target node name [%s]",
```
Comment (grcevski) [inline x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/PutShutdownNodeAction.java:160]: Same here, we don't need to concatenate the strings with `+`

**Review by grcevski — APPROVED** [review]
LGTM! I left two minor comments.

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Pull request merged**